apps/protocol-api/src/prisma/resolvers/Contribution.ts

```diff
@@ -545,18 +545,30 @@ export class ContributionCustomResolver {
     // grouping by a derived field not yet supported in prisma
     // would need to group by date, not the datetime as is stored in postg*/
     // YYY-MM-DD hh:mm:ss.sss
-    const user_id = args.where.id;
+    const userId = args.where.id;
     const start = args.where.startDate;
     const end = args.where.endDate;
 
     let guildWhere = Prisma.sql`gc."guild_id" is NULL`;
-    const guildIds = [...args.where?.guildIds];
-    if (guildIds.length > 0 && !args.where?.excludeUnassigned) {
-      guildWhere = Prisma.sql`(gc."guild_id" in (${Prisma.join(
-        guildIds,
-      )})  OR gc."guild_id" is NULL)`;
-    } else if (guildIds.length > 0 && args.where?.excludeUnassigned) {
-      guildWhere = Prisma.sql`(gc."guild_id" in (${Prisma.join(guildIds)}))`;
+
+    if (args.where.guildIds){
+      const guildIds = [...args.where.guildIds];
+      if (guildIds.length > 0 && !args.where?.excludeUnassigned) {
+        guildWhere = Prisma.sql`(gc."guild_id" in (${Prisma.join(
+          guildIds,
+        )})  OR gc."guild_id" is NULL)`;
+      } else if (guildIds.length > 0 && args.where?.excludeUnassigned) {
+        guildWhere = Prisma.sql`(gc."guild_id" in (${Prisma.join(guildIds)}))`;
+      }
+    }
+
+    let userWhere = Prisma.sql`
+      gc."user_id" = ${userId} OR gc."user_id" is null
+    `; 
+    if (userId == null) {
+      userWhere = Prisma.sql`
+        ${userWhere} OR TRUE
```

Comment:
My SQL syntax may be rusty, but shouldn't this be `gc."user_id" = ${userId} OR gc."user_id" is null OR  gc."user_id" is true`? Does OR with IS work?

Reply:
the intent here is to have the user clause evaluate to TRUE for all records if the `userId` is not supplied. 

I think the SQL query evaluates to `gc."user_id" = ${userId} OR gc."user_id" is null OR TRUE`, right? The last `OR TRUE` brings the disjunction high for every record